src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -202,6 +202,36 @@ private static T ConvertFromInvokeBindingResponse<T>(Autogenerated.InvokeBinding
             return response.Data.Value.IsEmpty ? default : TypeConverters.FromAny<TResponse>(response.Data, this.jsonSerializerOptions);
         }
 
+        public override async ValueTask<IEnumerable<GetBulkItem>> GetBulkStateAsync(string storeName, IEnumerable<string> keys, int parallelism = -1, CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(storeName, nameof(storeName));
+            if (keys.Count() == 0)
```

Comment:
It's generally not a great idea to walk an IEnumerable twice (here and L216), because you don't know what the code that generated it does 😆 . Do we have any idea how big we expect these lists of keys be to be? 

Some other options:
- Accept a data type that supports random access instead of IEnumerable
- Buffer the items here
- add the items to `envelope.Keys` and then check the count

Reply:
Changed to `IList<>`